### ISSUE TITLE
Geocode cities

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-server#758]
Send back coordinates for the cities JSONS. This would enable us to make a made near me for cosmetics.

https://world.openbeautyfacts.org/cities
https://world.openfoodfacts.org/cities
https://world.openbeautyfacts.org/cities.json

**Comment by CloCkWeRX** [comment]
It might be worth linking cities to their wikidata:
https://www.wikidata.org/wiki/Q213294 -> https://world.openfoodfacts.org/city/clamecy-nievre-france

That gives you coordinates, but it also gives you rich data *and* nice cross links to openstreetmap, etc.

**Comment by hangy** [comment]
Yup, cities need to be introduced with a useful taxonomy instead of the way they're currently extracted for French pacakger codes. Basically, at least packaging codes, origin of ingredients, and manufacturing/processing places link to geographical regions or cities in very similar ways.
Also discussed in #392 and linked issues. 😉

**Comment by teolemon** [comment]
http://www.geonames.org/export/

**Comment by CloCkWeRX** [comment]
Some possibilities here:
https://nominatim.org/release-docs/latest/api/Search/ - Can Nominatin suggest Town  / City / Suburb level results for things like "Location purchases" + recommend things with wikidata ids?

https://www.wikidata.org/wiki/Wikidata:SPARQL_query_service/queries/examples#Cities - hit the wikipdata sparql api endpoint for cities/countries?

**Comment by CloCkWeRX** [comment]
Actually, maybe worth superceding similar to #393 ?

**Comment by teolemon** [comment]
Open Prices has a partial solution for this. We could end up delegating location to the Open Prices backend (at least for "places where sold"
